apps/prairielearn/src/pages/instructorCourseAdminStaff/instructorCourseAdminStaff.html.ts

```diff
@@ -0,0 +1,748 @@
+                  </button>
+                </div>
+              </div>
+            </div>
+            ${StaffTable({
+              csrfToken: resLocals.__csrf_token,
+              courseUsers,
+              authnUser: resLocals.authn_user,
+              user: resLocals.user,
+              isAdministrator: resLocals.is_administrator,
+            })}
+            <small class="card-footer">
+              ${hasUnknownUsers
+                ? html`
+                    <p class="alert alert-warning">
+                      Users with name "<span class="text-danger">Unknown user</span>" either have
+                      never logged in or have an incorrect UID.
+                    </p>
+                  `
+                : ''}
+              <details>
+                <summary>Recommended access levels</summary>
+                ${AccessLevelsTable()}
+              </details>
+            </small>
+          </div>
+        </main>
+      </body>
+    </html>
+  `.toString();
+}
+
+function coursePermissionsRemoveStudentDataAccessForm({
+  csrfToken,
+  id,
+}: {
+  csrfToken: string;
+  id: string;
+}) {
+  return html`
+    <form name="remove-student-data-access" method="POST">
+      <input type="hidden" name="__action" value="remove_all_student_data_access" />
+      <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+
+      <div class="form-group mb-4">
+        <p class="form-text">
+          Taking this action will remove all student data access from all users (but will leave
+          these users on the course staff).
+        </p>
+      </div>
+
+      <div class="text-right mt-4">
+        <button type="button" class="btn btn-secondary" onclick="$('#${id}').popover('hide')">
+          Cancel
+        </button>
+        <button type="submit" class="btn btn-primary">Remove all student data access</button>
+      </div>
+    </form>
+  `;
+}
+
+function coursePermissionsDeleteNoAccessForm({ csrfToken, id }: { csrfToken: string; id: string }) {
+  return html`
+    <form name="delete-no-access" method="POST">
+      <input type="hidden" name="__action" value="delete_no_access" />
+      <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+
+      <div class="form-group mb-4">
+        <p class="form-text">
+          Taking this action will remove every user from course staff who has neither course content
+          access nor student data access.
+        </p>
+      </div>
+
+      <div class="text-right mt-4">
+        <button type="button" class="btn btn-secondary" onclick="$('#${id}').popover('hide')">
+          Cancel
+        </button>
+        <button type="submit" class="btn btn-primary">Delete users with no access</button>
+      </div>
+    </form>
+  `;
+}
+
+function coursePermissionsInsertForm({
+  csrfToken,
+  id,
+  uidsLimit,
+  courseInstances,
+}: {
+  csrfToken: string;
+  id: string;
+  uidsLimit: number;
+  courseInstances: any[];
```

Comment:
You can type this much more strongly:

```suggestion
  courseInstances: CourseInstance[];
```